### DISCUSSION TITLE
In port ESP32, why is pin GPIO 20 not valid?

### DISCUSSION THREAD

**Discussion by mitchb1** [micropython/micropython#9375] [Hardware & Peripherals] [answered]
In [/micropython/ports/esp32/machine_pin.c](https://github.com/micropython/micropython/blob/673db939b5c1f42c9411f684343d39811b7a16b2/ports/esp32/machine_pin.c#L162) GPIO 20 is null. Why is that? On [Adafruit ESP32 Feather V2](https://learn.adafruit.com/assets/112834) it is used for I2C SCL.

**Comment by jimmo** [comment, marked as answer]
The line you mentioned explains what's going on:

```c
    #if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 3, 2)
    {{&machine_pin_type}, GPIO_NUM_20},
    #else
    {{NULL}, -1},
    #endif
```

It needs to be compiled with IDF 4.4+

Unfortunately we still use IDF 4.2 for the releases due to an issue with memory fragmentation on the higher IDF versions. You can compile it yourself though, and as long as you're not using SSL it will be fine. (More info at #8940)
